### PR TITLE
transgui: patch to circumvent v3 rpc issue

### PR DESCRIPTION
**Pull request by RamKromberg**
###### Motivation for this change

https://github.com/NixOS/nixpkgs/issues/97430

###### Things done

reviewed the source and read up on the rpc; added the patch; built and executed on nixos 20.09; tried out adding, removing and moving torrents to verify nothing gets accidentally deleted...

I'm still not too happy about what essentially comes down to circumventing rpc changes between versions instead of properly updating the sources to the new specs but I haven't noticed anything that can lead to data loss and upstream hasn't reviewed much other than translations for months AND this project has a long history of going out of maintenance and forking...

### CODE REVIEW

**Comment by RamKromberg** [comment]
switched from 1329 to 1354 due to https://github.com/transmission-remote-gui/transgui/pull/1354#issuecomment-730876489

File: pkgs/applications/networking/p2p/transgui/default.nix [hunk 1]
```diff
@@ -24,6 +24,8 @@ stdenv.mkDerivation rec {
     -lgdk_pixbuf-2.0 -lpango-1.0 -latk-1.0 -lcairo -lc -lcrypto
   ";
 
+  patches = [ ./1354.patch ]; #TDDO: remove when transgui updates for transmission-daemon v3 rpc protocol
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/p2p/transgui/default.nix:27]: Please use fetchpatch instead.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by RamKromberg** [comment]
sorry. I had a cold. :parrot:

Anyhow, should be up-to-date and patched up for the not-so-new v.3 daemon rpc protocol.

File: pkgs/applications/networking/p2p/transgui/default.nix [hunk 2]
```diff
@@ -1,28 +1,34 @@
...
+    sha256 = "1x9wzii3q9zanpik4xc99jqsfrqch8vjmlx12jrvczxcfy51b1ba";
   };
 
+  patches = [
+    (fetchpatch {
+      url =
+        "https://patch-diff.githubusercontent.com/raw/transmission-remote-gui/transgui/pull/1354.patch";
+      sha256 = "0w2x7gcxp5kqczdz7ckfqhdz9hhkm62k8gcws54d6km7x9vc1023";
+    })
+  ]; # TDDO: remove when transgui updates for transmission-daemon v3 rpc protocol
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/p2p/transgui/default.nix:21]:
```suggestion
    # TDDO: remove when transgui updates for transmission-daemon v3 rpc protocol
    (fetchpatch {
      url = "https://github.com/transmission-remote-gui/transgui/commit/9275c3fb877dd753a1940d1b900630cdc09a0cc2.patch";
      sha256 = "0w2x7gcxp5kqczdz7ckfqhdz9hhkm62k8gcws54d6km7x9vc1023";
    })
  ];
```

File: pkgs/applications/networking/p2p/transgui/default.nix [hunk 3]
```diff
@@ -1,28 +1,34 @@
...
 
-  NIX_LDFLAGS = "
-    -L${stdenv.cc.cc.lib}/lib
-    -lX11 -lglib-2.0 -lgtk-x11-2.0 -lgdk-x11-2.0
-    -lgdk_pixbuf-2.0 -lpango-1.0 -latk-1.0 -lcairo -lc -lcrypto
-  ";
+  NIX_LDFLAGS = ''
+    -L${stdenv.cc.cc.lib}/lib -lX11 -lglib-2.0 -lgtk-x11-2.0
+    -lgdk-x11-2.0 -lgdk_pixbuf-2.0 -lpango-1.0 -latk-1.0 -lcairo
+    -lc -lcrypto'';
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/p2p/transgui/default.nix:31]:
```suggestion
    -L${stdenv.cc.cc.lib}/lib -lX11 -lglib-2.0 -lgtk-x11-2.0
    -lgdk-x11-2.0 -lgdk_pixbuf-2.0 -lpango-1.0 -latk-1.0 -lcairo
    -lc -lcrypto
  '';
```
Is there no easier way to do this?

File: pkgs/applications/networking/p2p/transgui/default.nix [hunk 4]
```diff
@@ -1,28 +1,34 @@
...
 
   prePatch = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/p2p/transgui/default.nix:33]:
```suggestion
  postPatch = ''
```

File: pkgs/applications/networking/p2p/transgui/default.nix [hunk 5]
```diff
@@ -33,18 +39,14 @@ stdenv.mkDerivation rec {
...
 
   LCL_PLATFORM = "gtk2";
 
-  desktopItem = makeDesktopItem rec {
-    name = "transgui";
-    exec = name + " %U";
-    icon = name;
+  desktopItem = makeDesktopItem {
+    name = pname;
+    exec = pname + " %U";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/p2p/transgui/default.nix:48]:
```suggestion
    exec = "${pname} %U";
```

File: pkgs/applications/networking/p2p/transgui/default.nix (hunk 3 above)
Comment (RamKromberg) [reply pkgs/applications/networking/p2p/transgui/default.nix:31]: I didn't get much luck with lazbuild but I'm not a pascal / lazarus dev so it's possible I'm missing something obvious.

**Comment by RamKromberg** [comment]
made the changes requested, updated and rebased on current master.

**Comment by RamKromberg** [comment]
btw, I'm not sure what to make of the farm's build failure as it builds fine on my system.

**Comment by SuperSandro2000** [comment]
It can't link on aarch64-linux.

```
make[4]: Entering directory '/build/fpcbuild-3.2.0/fpcsrc/packages/fpmkunit'
/nix/store/n35z01fwkf0alzn5igg5l4phc9r8gdk6-coreutils-9.0/bin/mkdir -p units_bs/aarch64-linux
/build/fpcbuild-3.2.0/fpcsrc/compiler/ppca64 src/fpmkunit.pp  -Fu/build/fpcbuild-3.2.0/fpcsrc/rtl/units/aarch64-linux -FUunits_bs/aarch64-linux -Fu../paszlib/src -Fu../hash/src -Fi../paszlib/src  -Fi../fcl-process/src/unix -Fu../fcl-process/src -Fi../fcl-process/src/linux -Fi../fcl-process/src/dummy -Fu../libtar/src 
make[4]: Leaving directory '/build/fpcbuild-3.2.0/fpcsrc/packages/fpmkunit'
/build/fpcbuild-3.2.0/fpcsrc/compiler/ppca64 fpmake.pp  -Fu/build/fpcbuild-3.2.0/fpcsrc/packages/fpmkunit/units_bs/aarch64-linux -Fu/build/fpcbuild-3.2.0/fpcsrc/rtl/units/aarch64-linux  
/nix/store/vwsfkvn1zp4z0gznxx2r478v0rrwncpk-binutils-2.38/bin/ld: /build/fpcbuild-3.2.0/fpcsrc/rtl/units/aarch64-linux/cprt0.o: in function `_start':
(.text+0x54): undefined reference to `__libc_csu_init'
/nix/store/vwsfkvn1zp4z0gznxx2r478v0rrwncpk-binutils-2.38/bin/ld: (.text+0x58): undefined reference to `__libc_csu_init'
/nix/store/vwsfkvn1zp4z0gznxx2r478v0rrwncpk-binutils-2.38/bin/ld: (.text+0x5c): undefined reference to `__libc_csu_fini'
/nix/store/vwsfkvn1zp4z0gznxx2r478v0rrwncpk-binutils-2.38/bin/ld: (.text+0x60): undefined reference to `__libc_csu_fini'
fpmake.pp(55) Error: Error while linking
fpmake.pp(55) Fatal: There were 1 errors compiling module, stopping
Fatal: Compilation aborted
make[3]: *** [Makefile:1725: fpmake] Error 1
make[3]: Leaving directory '/build/fpcbuild-3.2.0/fpcsrc/packages'
make[2]: *** [Makefile:2681: packages_smart] Error 2
make[2]: Leaving directory '/build/fpcbuild-3.2.0/fpcsrc'
make[1]: *** [Makefile:2836: build-stamp.aarch64-linux] Error 2
make[1]: Leaving directory '/build/fpcbuild-3.2.0/fpcsrc'
make: *** [Makefile:2307: fpcsrc/build-stamp.aarch64-linux] Error 2
```

File: pkgs/applications/networking/p2p/transgui/default.nix [hunk 6]
```diff
@@ -1,30 +1,38 @@
...
-    sha256 = "1dyx778756zhvz5sxgdvy49p2c0x44w4nmcfd90wqrmgfknncnf5";
+    rev = "0e2c2a07c1b21b1704c0a4945a111a8aa1050a1a";
+    sha256 = "1x9wzii3q9zanpik4xc99jqsfrqch8vjmlx12jrvczxcfy51b1ba";
   };
 
+  patches = [
+    # TDDO: remove when transgui updates for transmission-daemon v3 rpc protocol
+    (fetchpatch {
+      url =
+        "https://patch-diff.githubusercontent.com/raw/transmission-remote-gui/transgui/pull/1354.patch";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/p2p/transgui/default.nix:19]:
```suggestion
      url = "https://github.com/transmission-remote-gui/transgui/commit/9275c3fb877dd753a1940d1b900630cdc09a0cc2.patch";
```

File: pkgs/applications/networking/p2p/transgui/default.nix [hunk 7]
```diff
@@ -1,30 +1,38 @@
...
+      sha256 = "0w2x7gcxp5kqczdz7ckfqhdz9hhkm62k8gcws54d6km7x9vc1023";
+    })
+  ];
+
   nativeBuildInputs = [ pkg-config unzip ];
-  buildInputs = [
-    fpc lazarus stdenv.cc
-    libX11 glib gtk2 gdk-pixbuf pango atk cairo openssl
+  buildInputs = [ fpc lazarus stdenv.cc libX11 glib gtk2 gdk-pixbuf
+    pango atk cairo openssl
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/p2p/transgui/default.nix:26]:
```suggestion
  buildInputs = [
    fpc lazarus stdenv.cc libX11 glib gtk2 gdk-pixbuf
    pango atk cairo openssl
```

File: pkgs/applications/networking/p2p/transgui/default.nix [hunk 8]
```diff
@@ -64,10 +68,12 @@ stdenv.mkDerivation rec {
   '';
 
   meta = {
-    description = "A cross platform front-end for the Transmission Bit-Torrent client";
+    description =
+      "A cross platform front-end for the Transmission Bit-Torrent client";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/p2p/transgui/default.nix:72]:
```suggestion
    description = "A cross platform front-end for the Transmission Bit-Torrent client";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
.

File: pkgs/applications/networking/p2p/transgui/default.nix (hunk 8 above)
Comment (RamKromberg) [reply pkgs/applications/networking/p2p/transgui/default.nix:72]: yeah I hate the formatting nixfmt had me use but I assumed it was a strict thing... I'll be sure to stop using and stick to my taste from now on :)

File: pkgs/applications/networking/p2p/transgui/default.nix (hunk 6 above)
Comment (RamKromberg) [reply pkgs/applications/networking/p2p/transgui/default.nix:19]: sure.

File: pkgs/applications/networking/p2p/transgui/default.nix (hunk 7 above)
Comment (RamKromberg) [reply pkgs/applications/networking/p2p/transgui/default.nix:26]: k

File: pkgs/applications/networking/p2p/transgui/default.nix (hunk 8 above)
Comment (SuperSandro2000) [reply pkgs/applications/networking/p2p/transgui/default.nix]: Use nixpkgs-fmt which does not doe such changes.

File: pkgs/applications/networking/p2p/transgui/default.nix [hunk 9]
```diff
@@ -1,30 +1,39 @@
...
-    sha256 = "1dyx778756zhvz5sxgdvy49p2c0x44w4nmcfd90wqrmgfknncnf5";
+    rev = "0e2c2a07c1b21b1704c0a4945a111a8aa1050a1a";
+    sha256 = "1x9wzii3q9zanpik4xc99jqsfrqch8vjmlx12jrvczxcfy51b1ba";
   };
 
+  patches = [
+    # TDDO: remove when transgui updates for transmission-daemon v3 rpc protocol
+    (fetchpatch {
+      url =
+        "https://github.com/transmission-remote-gui/transgui/commit/9275c3fb877dd753a1940d1b900630cdc09a0cc2.patch";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/p2p/transgui/default.nix:19]:
```suggestion
      url = "https://github.com/transmission-remote-gui/transgui/commit/9275c3fb877dd753a1940d1b900630cdc09a0cc2.patch";
```

Comment (RamKromberg) [reply pkgs/applications/networking/p2p/transgui/default.nix]: k

**Comment by RamKromberg** [comment]
made the final requested change to the patch url and corrected "Bit-Torrent" into "BitTorrent" in the meta.description while at it.

**Comment by RamKromberg** [comment]
Thanks and sorry it took so long :)

**Pull request merged**